transcendence-app/src/shared/db/base.repository.ts

```diff
@@ -0,0 +1,18 @@
+import { ID } from './models';
+
+export type DataResponseWrapper<T> = {
+  data: T[] | null;
+  entries: number | null;
+  error: string | null;
+};
+
+export interface IBaseRepository<T> {
+  getByKey(key: string, id: ID): Promise<DataResponseWrapper<T>>;
+  deleteByKey(key: string, id: ID): Promise<DataResponseWrapper<T>>;
+  add(e: T): Promise<DataResponseWrapper<T>>;
```

Comment:
> What is the e parameter of this method?

the `e` parameter stands for entity.

> Could we change it to be of type Partial<T>, or is it a bad idea?

I believe no, the entity itself already matches table requirements. For example, you could have nullable/optional entries in your table, this is already represented by the entity object. If you add `Partial` that would mean you'd allow some of required fields not to be passed and, since this method is for adding new entries to the db, that is no acceptable.

Reply:
I think dealing with optional parameters could be solved with [`CONSTRAINS`](https://www.postgresql.org/docs/14/ddl-constraints.html)(for example, with the `NOT NULL` check of some columns from the `users` table)

`Partial<T>`, in my opinion, looks a good enough type for a generic :sweat_smile:.

Reply:
I am resolving this conversation and moving it to this related [thread](https://github.com/42AGV/ft_transcendence/pull/51#discussion_r895067913)